### DISCUSSION TITLE
Border colors

### DISCUSSION THREAD

**Discussion by BigfootN** [MunifTanjim/nui.nvim#358] [Q&A] [answered]
Hello,

I would like to set border colors but it seems like `nvim_set_hl` doesn't seem to affect the `popup`. I'm using `noice.nvim` but the backend is `nui`.

Here is my configuration:

```lua
return {
    "folke/noice.nvim",
    dependencies = {
        "MunifTanjim/nui.nvim"
    },
    opts = {
        cmdline = {
            enabled = true,
            view = 'cmdline_popup',
        }
    },
    config = function(_, opts)
        vim.api.nvim_set_hl(0, 'FloatBorder', { bg = '#3B4252', fg = '#5E81AC' })
        vim.api.nvim_set_hl(0, 'NormalFloat', { bg = '#3B4252' })
        require("noice").setup(opts)
        vim.api.nvim_set_hl(0, 'FloatBorder', { bg = '#3B4252', fg = '#5E81AC' })
        vim.api.nvim_set_hl(0, 'NormalFloat', { bg = '#3B4252' })
    end
}
```

The color is the same as with the default configuration.

Thank you very much in advance for any help

**Comment by MunifTanjim** [comment]
You should ask this in `noice.nvim` repo. I don't know how `noice.nvim` uses highlight groups... It doesn't necessarily has to be the default `FloatBorder`/`NormalFloat`. Also the first parameter f `nvim_set_hl` is the namespace. `noice.nvim` could be using a different namespace (not the default `0`).

**Reply by BigfootN** [reply, marked as answer]
Thank you for your quick response and help.

I finally figured it out. For anyone interested, the [documentation of noice](https://github.com/folke/noice.nvim/blob/main/doc/noice.nvim.txt#L794) clearly indicates the used highlight groups.

So this example works:

```lua
config = function(_, opts)
    require("noice").setup(opts)
    vim.api.nvim_set_hl(0, 'NoiceCmdlinePopupBorderCmdline', { fg = "#e78a4e" })
end
```

Thank you for your time and sorry for wasting it.
